File: json/perl/delete/json_delete.pl
#! /usr/bin/perl
#
#	json_delete.pl
#
#					Jun/22/2020
#
#
# ----------------------------------------------------------------
use	strict;
use	warnings;
use	utf8;
#
use Encode;
use JSON;
#
use lib '/var/www/data_base/common/perl_common';
use text_manipulate;
use file_io;
#
# ----------------------------------------------------------------
print STDERR encode_utf8 ("*** 開始 ***\n");
my $file_json= $ARGV[0];
my $id_in = $ARGV[1];
#
print	$id_in . "\n";
#
my $str_json = file_io::string_read_proc ($file_json);
#
my %dict_aa = %{decode_json ($str_json)};
#
%dict_aa = text_manipulate::dict_delete_proc ($id_in,%dict_aa);
#
my $json_str_new = encode_json (\%dict_aa);
#
#print $json_str_new;
#
file_io::string_write_proc ($file_json,$json_str_new);
#
print STDERR encode_utf8 ("*** 終了 ***\n");
#
# ----------------------------------------------------------------

File: json/narwhal/delete/json_delete.js
#! /usr/local/bin/narwhal
// ------------------------------------------------------------------
//
//	json_delete.js
//
//						May/07/2011
//
// ------------------------------------------------------------------
var text_manipulate = require ('/var/www/data_base/common/narwhal_common/text_manipulate.js');
// ------------------------------------------------------------------

print	("*** 開始 ***");
var args = require('system').args;
var file_json = args[1];
var id_in = args[2];

print	("id_in = " + id_in);

var json_str = readFile (file_json);

var dict_aa = JSON.parse (json_str);

delete dict_aa[id_in];

var out_str = JSON.stringify (dict_aa);

text_manipulate.string_write_proc (file_json,out_str);

print	("*** 終了 ***");
// ------------------------------------------------------------------
